vms/platformvm/dao/dao.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2022, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package dao
+
+import (
+	"errors"
+	"time"
+
+	"github.com/chain4travel/caminogo/codec"
+)
+
+var (
+	errStartTimeBeforeEndTime = errors.New("startTime must be before endTime")
+	errNonPositiveThreshold   = errors.New("the number of votes to accept a proposal must be at least 1")
+
+	Codec codec.Manager
+)
+
+type ProposalState int
+
+const (
+	ProposalStateUnknown ProposalState = iota
+	ProposalStatePending
+	ProposalStateAccepted
+	ProposalStateConcluded
+)
+
+func (p ProposalState) String() string {
+	switch p {
+	case ProposalStatePending:
+		return "Pending"
+	case ProposalStateAccepted:
+		return "Accepted"
+	case ProposalStateConcluded:
+		return "Concluded"
+	default:
+		return "Unknown"
+	}
+}
+
+// Dao ProposalConfiguration.
+type ProposalConfiguration struct {
+	// The threshold of votes which has to be reached
+	Thresh uint32 `serialize:"true" json:"threshold"`
+
+	// Unix time this proposal starts
+	Start uint64 `serialize:"true" json:"startTime"`
+
+	// Unix time this Dao finishes
+	End uint64 `serialize:"true" json:"endTime"`
+}
+
+func HasConcluded(state ProposalState) bool {
+	return state == ProposalStateConcluded
+}
+
+// StartTime is the time that this validator will enter the validator set
```

Comment:
this documentation is obviously copied from validator.go and does not describe its real purpose